parse/SpeciesParser.cpp

```diff
@@ -276,17 +279,82 @@ namespace {
         parse::detail::planet_type_parser_rules planet_type_rules;
         parse::detail::planet_environment_parser_rules planet_environment_rules;
     };
+
+    using manifest_start_rule_signature = void (std::vector<std::string>&);
+
+    struct manifest_grammar : public parse::detail::grammar<manifest_start_rule_signature> {
+        manifest_grammar(const parse::lexer& tok,
+                         const std::string& filename,
+                         const parse::text_iterator& first, const parse::text_iterator& last) :
+            manifest_grammar::base_type(start),
+            labeller(tok)
+        {
+            namespace phoenix = boost::phoenix;
+            namespace qi = boost::spirit::qi;
+
+            using phoenix::push_back;
+
+            qi::_1_type _1;
+            qi::_2_type _2;
+            qi::_3_type _3;
+            qi::_4_type _4;
+            qi::_r1_type _r1;
+
+            species_manifest
+                =    tok.SpeciesCensusOrdering_
+                >    *(labeller.rule(Tag_token)       > tok.string [ push_back(_r1, _1) ])
+                ;
+
+            start
+                =   +species_manifest(_r1)
+                ;
+
+            species_manifest.name("ParsedSpeciesCensusOrdering");
+
+#if DEBUG_PARSERS
+            debug(species_manifest);
+#endif
+
+            qi::on_error<qi::fail>(start, parse::report_error(filename, first, last, _1, _2, _3, _4));
+        }
+
+        using manifest_rule = parse::detail::rule<void (std::vector<std::string>&)>;
+        using start_rule = parse::detail::rule<manifest_start_rule_signature>;
+
+        parse::detail::Labeller labeller;
+        manifest_rule species_manifest;
+        start_rule start;
+    };
 }
 
 namespace parse {
     start_rule_payload species(const boost::filesystem::path& path) {
         const lexer lexer;
-        start_rule_payload species_;
+        start_rule_payload::first_type species_;
+        start_rule_payload::second_type ordering;
+
+        boost::filesystem::path manifest_file;
 
         for (const boost::filesystem::path& file : ListScripts(path)) {
-            /*auto success =*/ detail::parse_file<grammar, start_rule_payload>(lexer, file, species_);
+            if (file.filename() == "SpeciesCensusOrdering.focs.txt" ) {
```

Comment:
Can we use .inf for such files, or should we rename the existing .inf files?

Reply:
I think the distinction is that .inf files shouldn't contain scripted content... though in that case, monster fleets and categories should be in .focs.txt rather than .inf files. But the list in this pull request would fit in a .inf.

Regardless, lists like this, such as https://github.com/freeorion/freeorion/blob/master/default/scripting/starting_unlocks/items.inf

have no initial header line. The header is unnecessary if the file can only contain one top-level entry, so I think there is no need for `SpeciesCensusOrdering` in the file or parser.

Reply:
@geoffthemedio 

Noted.  The implementation in this PR is pretty much a copy&paste job from the current ShipDesignParser, which uses a similar manifest file introduced with commit 36d659be52ba5e494a4c206b67e58b9ae1398c50

Reply:
Fair enough. My interpretation of .inf vs. .focs.txt is a bit spectulative anyway...

Reply:
So should this be rewritten to use '*.inf' semantics or not?

Reply:
I'm not sure. There are examples using both formats. Since I'm not sure, and nobody else who has a strong opinion has commented, I suppose it's fine to leave as you have it.